site/formulas/decoding.md

```diff
@@ -7,14 +7,14 @@ follows.
 1. Check that `s_bytes` is the canonical encoding of a field element, or else abort.
 2. Decode `s_bytes`, a byte-string, into \\(s\\), a field element.
 3. Check that \\(s\\) is non-negative, or else abort.
-4. \\(u_1 \gets 1 - a s^{2} \\).
-5. \\(u_2 \gets 1 + a s^{2} \\).
+4. \\(u_1 \gets 1 + a s^{2} \\).
+5. \\(u_2 \gets 1 - a s^{2} \\).
 6. \\(v \gets adu_1^2 - u_2^2 \\).
-7. \\(I \gets \mathrm{invsqrt}( v u_1^{2} ) \\), or abort if the square root does not exist.
+7. \\(I \gets \mathrm{invsqrt}( v u_2^{2} ) \\), or abort if the square root does not exist.
 8. \\(D_x \gets I u_2 \\).
 9. \\(D_y \gets I D_x v \\).
 10. \\(x \gets |2 s D_x| \\), i.e., compute \\(2sD_x\\) and negate it if it is negative.
-11. Compute \\(y \gets u_2 D_y \\).
+11. Compute \\(y \gets u_1 D_y \\).
```

Comment:
This should also be `u_2` here. The [code](https://github.com/dalek-cryptography/curve25519-dalek/blob/39145da396226516d8d1f663ad26f86bf04856db/src/ristretto.rs#L279) is:

```
        // y == (1-as²)/(1+as²)
        let y = &u2 * &Dy;
```

(And again the comment in the code is wrong.)